frontend/components/common/Table.tsx

```diff
@@ -217,106 +233,121 @@ const Table = ({
-                          }
-                        : null
-                    }
-                    styles={styles}
-                    components={components}
-                    onChange={(value) =>
-                      handleFilterChange({
-                        value: value || null,
-                        label: filterOption.label ? filterOption.label : null,
-                      })
-                    }
-                    isClearable={true}
-                    placeholder={`Filter by ${titleize(filterOption.label)}`}
-                    options={filterOption.options.map((option) => {
-                      return { value: option.key, label: option.label }
-                    })}
-                  />
-                </div>
-              ))}
-          </div>
+        <div className="is-pulled-left is-clearfix" style={{ width: '70%' }}>
+          {filterOptions &&
+            filterOptions.map((filterOption) => (
+              <span
+                style={{ marginRight: '10px', width: '40%', float: 'left' }}
+              >
+                <Select
+                  value={
+                    selectedFilter[filterOption.label]
+                      ? {
+                          label: selectedFilter[filterOption.label].label,
+                          value: selectedFilter[filterOption.label].key,
+                        }
+                      : null
+                  }
+                  styles={styles}
+                  components={components}
+                  onChange={(value) =>
+                    handleFilterChange({
+                      value: value || null,
+                      label: filterOption.label ? filterOption.label : null,
+                    })
+                  }
+                  isClearable={true}
+                  placeholder={`Filter by ${titleize(filterOption.label)}`}
+                  options={filterOption.options.map((option) => {
+                    return { value: option.key, label: option.label }
+                  })}
+                />
+              </span>
+            ))}
         </div>
       </Toolbar>
 
-      {tableData.length > 0 ? (
+      {headerGroups.length > 0 ? (
         <table className="table is-fullwidth" {...getTableProps()}>
           <thead>
             {headerGroups.map((headerGroup) => (
               <tr {...headerGroup.getHeaderGroupProps()}>
                 {headerGroup.headers.map((column) => (
                   <th {...column.getHeaderProps(column.getSortByToggleProps())}>
-                    {titleize(column.render('Header'))}
-                    <span style={{ marginLeft: '1rem' }}>
-                      {column.isSorted ? (
-                        column.isSortedDesc ? (
-                          <Icon name="chevron-down" />
+                    <div onClick={() => handleColumnsSort(column.Header)}>
+                      {titleize(column.render('Header'))}
+                      <span style={{ marginLeft: '1rem' }}>
```

Comment:
It's hard to see which icons are being rendered in which conditions, so consider extracting this into a function that returns the icon name!
```javascript
const iconNameForSortedState = (column): string => {
    if(column.isSorted){
        if(column.isSortedDesc) {
            return "triangle-down"
        }else{
            return "triangle-up"
        }
    }
    return "group-arrows"
}
```

Reply:
This might still not be easy to do, in one of the cases I will be returning an empty string and not an icon.